src/index.js

```diff
@@ -68,21 +68,27 @@ class GoogleCloudStorageAdapter extends BaseAdapter {
   }
 
   read(options = {}) {
-    const readStream = this.bucket.file(options.path).createReadStream()
+    const prefix = `https://${this.assetDomain}/`
+    let path = options.path
+    if (path.indexOf(prefix) === 0) {
+      path = path.slice(prefix.length);
+    }
+    
+    const readStream = this.bucket.file(path).createReadStream()
     let contents = null;
     return new Promise(function(resolve, reject) {
       readStream.on('error', function(err) {
         return reject(err);
       });
       readStream.on('data', function(data) {
         if (contents) {
-          contents = data;
-        } else {
           contents = Buffer.concat([contents, data]);
+        } else {
+          contents = data;
```

Comment:
Hi @cheungpat, what is the reason for this change? This changes the if...else block compared to what it currently is in `master`.

Reply:
The value of `contents` is initially null, so the `else` block is executed first, which initialize the value of `contents`.

If this change is not applied, `Buffer.concat([null, data])` is called, which will cause error.

Reply:
Ah of course, that makes sense and given your explanation below saying that `read()` is called rarely, it explains why this has not been identified as a bug.